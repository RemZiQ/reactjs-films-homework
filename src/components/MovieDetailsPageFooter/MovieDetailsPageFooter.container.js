import { connect } from 'react-redux';
import MovieDetailsPageFooter from './MovieDetailsPageFooter';


const mapStateToProps = store => ({ currentMovie: store.currentMovie });


export default connect(mapStateToProps, null)(MovieDetailsPageFooter);
