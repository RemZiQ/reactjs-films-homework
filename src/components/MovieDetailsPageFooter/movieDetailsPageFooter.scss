.movieDetailsPage__footer {
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 550px;
}
