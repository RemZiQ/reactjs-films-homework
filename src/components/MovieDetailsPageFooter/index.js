import MovieDetailsPageFooter from './MovieDetailsPageFooter';

export default MovieDetailsPageFooter;
