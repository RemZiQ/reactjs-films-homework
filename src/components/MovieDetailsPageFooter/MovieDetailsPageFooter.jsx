import React from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';

import StarRating from '../MovieDetailsPageStarRating';
import Action from '../MovieDetailsPageAction';
import './movieDetailsPageFooter.scss';

export const MovieDetailsPageFooter = ({ store }) => {
  if (store) {
    const { genres, id } = store;
    const rating = store.vote_average / 2;
    const hour = (store.runtime - (store.runtime % 60)) / 60;
    const duration = `${hour}h ${store.runtime - 60 * hour}m`;
    return (
      <div className="movieDetailsPage__footer">
        <StarRating
          title={store.title}
          genres={genres}
          duration={duration}
          rating={rating}
          id={id}
        />
        <Action info={store.overview} />
      </div>
    );
  }
  return (<div>Loading</div>);
};

MovieDetailsPageFooter.propTypes = {
  store: propTypes.objectOf(propTypes.any),
};

MovieDetailsPageFooter.defaultProps = {
  store: null,
};

const mapStateToProps = store => ({ store: store.currentMovie });


export default connect(mapStateToProps, null)(MovieDetailsPageFooter);
