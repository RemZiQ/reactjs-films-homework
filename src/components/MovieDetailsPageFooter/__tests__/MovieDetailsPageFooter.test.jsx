import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
// import { Provider } from 'react-redux';
// import { MovieDetailsPageFooter } from './MovieDetailsPageFooter';
import { MovieDetailsPageFooter } from '../MovieDetailsPageFooter';


it('renders well', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MovieDetailsPageFooter />);
  expect(result).toMatchSnapshot();
});

it('renders well2222', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<MovieDetailsPageFooter store={{}} />);
  expect(result).toMatchSnapshot();
});

// it('renders well', () => {
//   const renderer = new ShallowRenderer();
//   const result = renderer.render(<MovieDetailsPageFooter store={null} />);
//   expect(result).toMatchSnapshot();
// });
