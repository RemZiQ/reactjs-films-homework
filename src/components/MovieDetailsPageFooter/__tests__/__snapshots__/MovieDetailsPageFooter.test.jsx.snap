// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders well 1`] = `
<div>
  Loading
</div>
`;

exports[`renders well2222 1`] = `
<div
  className="movieDetailsPage__footer"
>
  <StarRating
    duration="NaNh NaNm"
    genres={
      Array [
        99,
      ]
    }
    id={null}
    rating={NaN}
    title="Sorry, no title here"
  />
  <Connect(Action) />
</div>
`;
